src/renderer/services/file-management-system/Md5Hasher.ts

```diff
@@ -0,0 +1,99 @@
+import * as CryptoJS from "crypto-js";
+
+/**
+ * Abstract hasher template.
+ */
+ interface Hasher {
+    /**
+     * The number of 32-bit words this hasher operates on. Default: 16 (512 bits)
+     */
+    blockSize: number;
+    /**
+     * Resets this hasher to its initial state.
+     *
+     * @example
+     *
+     *     hasher.reset();
+     */
+    reset(): void;
+    /**
+     * Updates this hasher with a message.
+     *
+     * @param messageUpdate The message to append.
+     *
+     * @return This hasher.
+     *
+     * @example
+     *
+     *     hasher.update('message');
+     *     hasher.update(wordArray);
+     */
+    update(messageUpdate: CryptoJS.lib.WordArray | string): this;
+    /**
+     * Finalizes the hash computation.
+     * Note that the finalize operation is effectively a destructive, read-once operation.
+     *
+     * @param messageUpdate (Optional) A final message update.
+     *
+     * @return The hash.
+     *
+     * @example
+     *
+     *     var hash = hasher.finalize();
+     *     var hash = hasher.finalize('message');
+     *     var hash = hasher.finalize(wordArray);
+     */
+    finalize(messageUpdate?: CryptoJS.lib.WordArray | string): CryptoJS.lib.WordArray;
+}
+
+export default class Md5Hasher{
+
+    public static deserialize (serialized_md5: string){
+        const hasher = CryptoJS.algo.MD5.create();
+        
+        /** Recursively copy properties from object source to object target. */
+        function restore_data(source: any, target: any) {
```

Comment:
Hm, are you saying that you think the deep copy is not necessary (your comment about removing those lines)?  Or just that we could use lodash to do the deep copy?

I tried `_.copyDeep`.  There is a difference in how the copy works - in this hand rolled version, the target starts off as a new Hasher.  Lodash doesn't take a target object, so I presume it starts with an empty object.  Lodash produces an object with missing members:

`TypeError: this.hasher.finalize is not a function`

I think improving this can be its own ticket.  Ok with you?

Reply:
Paired on refactoring this.